### PR TITLE
UI 35: added dropdown components + refactored button components

### PR DESCRIPTION
**Pull request by andrewtpham**
## What is the Purpose?
- added dropdown components
- refactors button components so you can customize size of button components

## Mentions?
@ninetteadhikari @chelliah

## Issue(s) affected?
#35

### CODE REVIEW

**Comment by andrewtpham** [comment]
@michaelbukachi would you mind reviewing this PR? Currently it may not be using the component as it should. I'm having issues trying to replace `<el-dropdown>` with `<basic-dropdown>` and dealing with the multiple slots. Thanks 🙏

File: src/components/dropdown/BasicDropdown.vue [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<template>
+    <el-dropdown>
+    <el-dropdown-menu slot="dropdown">
+        <el-dropdown-menu slot="dropdown">
+            <slot></slot>
+    </el-dropdown-menu>
+  </el-dropdown-menu>
+</el-dropdown>
```
Comment (michaelbukachi) [inline src/components/dropdown/BasicDropdown.vue:8]:
```
  <el-dropdown>
    <span class="el-dropdown-link">
        More options<i class="el-icon-arrow-down el-icon--right"></i>
    </span>
    <el-dropdown-menu slot="dropdown">
      <slot name="items"></slot>
    </el-dropdown-menu>
  </el-dropdown>
```

File: src/stories/Dropdown.stories.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        <span class="el-dropdown-link">
+        More options<i class="el-icon-arrow-down el-icon--right"></i>
+        </span>
+        <el-dropdown-menu slot="dropdown">
+            <el-dropdown-item>Edit</el-dropdown-item>
+            <el-dropdown-item>Change data type</el-dropdown-item>
+            <el-dropdown-item disabled>For admins only</el-dropdown-item>
+            <el-dropdown-item divided>Delete</el-dropdown-item>
+        </el-dropdown-menu>
+    </el-dropdown>
```
Comment (michaelbukachi) [inline src/stories/Dropdown.stories.js:24]:
```
    <basic-dropdown>
      <template slot="items">
        <el-dropdown-item>Edit</el-dropdown-item>
        <el-dropdown-item>Change data type</el-dropdown-item>
        <el-dropdown-item disabled>For admins only</el-dropdown-item>
        <el-dropdown-item divided>Delete</el-dropdown-item>
      </template>
    </basic-dropdown>
```

Comment (andrewtpham) [reply src/stories/Dropdown.stories.js:24]: Thanks @michaelbukachi if we remove the `<span>` and hard code it in the component, does that then mean we need to change the label in the component each time? I.e. "More options" label? Or can we expose that on the stories.js file as well?

Comment (michaelbukachi) [reply src/stories/Dropdown.stories.js:24]: We can use a prop to dynamically change the label.

**Comment by michaelbukachi** [comment]
Hey @henryohanga. The idea is to  have ready made components available for using without having to mess around with props unless it is absolutely necessary.

File: src/stories/Dropdown.stories.js (hunk 2 above)
Comment (andrewtpham) [reply src/stories/Dropdown.stories.js]: @michaelbukachi could you help suggest how I add a prop for the label?

Comment (andrewtpham) [reply src/stories/Dropdown.stories.js]: @michaelbukachi Actually, I was able to add an additional slot for the dropdown title. Let me know if this method is alright for exposing the label.

**Pull request merged**